pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -111,14 +111,17 @@ func (h *handler) nodesNeedUpgrade(cluster *mgmtv3.Cluster, version string) (boo
 		return false, err
 	}
 	for _, node := range v3NodeList {
-		isNewer, err := nodesyncer.IsNewerVersion(node.Status.InternalNodeStatus.NodeInfo.KubeletVersion, version)
-		if err != nil {
-			return false, err
-		}
-		if isNewer {
-			logrus.Debugf("[k3s-based-upgrader] cluster [%s] version [%s] is newer than observed node [%s] version [%s]",
-				cluster.Name, version, node.Name, node.Status.InternalNodeStatus.NodeInfo.KubeletVersion)
-			return true, nil
+		// if node is windows, skip upgrade check
+		if os, ok := node.Status.NodeLabels["kubernetes.io/os"]; ok && os != "windows" {
```

Comment:
I would've used the `core1.LabelOSStable` but it seems we use either across the codebase.

Reply:
Ah, I didn't know it was there in corev1. It's a better choice, so I'll update it.